test/test_jit.py

```diff
@@ -339,6 +341,23 @@ def doit(x, y):
         self.assertEqual(z, torch.sigmoid(torch.tanh(x * (x + y) + 1)))
         self.assertEqual(z, z2)
 
+    @unittest.skipIf(IS_WINDOWS, "NYI: fuser support for Windows")
+    @unittest.skipIf(not RUN_CUDA_MULTI_GPU, "needs non-zero device")
+    def test_compile_fuse_last_device(self):
```

Comment:
Can we already handle a situation where we have multiple equivalent traces but on different GPUs?

Reply:
Pretty sure that works because CompiledFunction specializes a trace based on the devices of the inputs. This just fixes the fact that the compiler needs to know which device a fusion group is run on.